lte/gateway/c/oai/tasks/gtpv1-u/gtpv1u_task.c

```diff
@@ -200,6 +200,33 @@ int gtpv1u_add_tunnel(
       ue, ue_ipv6, vlan, enb, i_tei, o_tei, imsi, flow_dl, flow_precedence_dl);
 }
 
+int gtpv1u_add_s8_tunnel(
+    struct in_addr ue, struct in6_addr* ue_ipv6, int vlan, struct in_addr enb,
+    struct in_addr pgw, uint32_t i_tei, uint32_t o_tei, Imsi_t imsi,
+    struct ip_flow_dl* flow_dl, uint32_t flow_precedence_dl) {
+  OAILOG_DEBUG(LOG_GTPV1U, "Add S8 tunnel ue %s", inet_ntoa(ue));
+  if (gtp_tunnel_ops->add_s8_tunnel) {
```

Comment:
Is function pointer, "add_s8_tunnel" initialized in gtp_tunnel_ops, because I didn't notice.
static const struct gtp_tunnel_ops openflow_ops = {
    .init                   = openflow_init,
    .uninit                 = openflow_uninit,
    .reset                  = openflow_reset,
    .add_tunnel             = openflow_add_tunnel,
    .del_tunnel             = openflow_del_tunnel,
    .discard_data_on_tunnel = openflow_discard_data_on_tunnel,
    .forward_data_on_tunnel = openflow_forward_data_on_tunnel,
    .add_paging_rule        = openflow_add_paging_rule,
    .delete_paging_rule     = openflow_delete_paging_rule,
    .send_end_marker        = openflow_send_end_marker,
    .get_dev_name           = openflow_get_dev_name,
};

Reply:
yes, its missing.